tests/src/testnonlinearelasticity.hh

```diff
@@ -72,45 +75,31 @@ auto NonLinearElasticityLoadControlNRandTR(const Material& mat) {
       dirichletFlags[localView.index(localIndex)] = true;
   });
 
-  auto sparseAssembler = SparseFlatAssembler(fes, dirichletValues);
+  auto sparseAssembler = makeSparseFlatAssembler(fes, dirichletValues);
 
   Eigen::VectorXd d;
   d.setZero(basis.flat().size());
   double lambda = 0.0;
 
-  auto req = FERequirements().addAffordance(Ikarus::AffordanceCollections::elastoStatics);
+  auto req = typename FEType::Requirement(d, lambda);
 
-  auto residualFunction = [&](auto&& disp_, auto&& lambdaLocal) -> auto& {
-    req.insertGlobalSolution(Ikarus::FESolutions::displacement, disp_)
-        .insertParameter(Ikarus::FEParameter::loadfactor, lambdaLocal);
-    return sparseAssembler.getVector(req);
-  };
+  sparseAssembler->bind(req, Ikarus::AffordanceCollections::elastoStatics);
+  auto nonLinOp = Ikarus::NonLinearOperatorFactory::op(sparseAssembler, EnforcingDBCOption::Reduced);
 
```

Comment:
`sparseAssembler->bind` has a default `EnforcingDBCOption qt = EnforcingDBCOption::Full`. This is then passed later to `trFactory.create(sparseAssembler)`. This means the nonlinear solver is not currently tested for `EnforcingDBCOption::Reduced` here right? This is the only test where this option is mentioned in combination with a nonlinear solver

Reply:
I refactored the initial tests in such a way that they have the same behaviour therefore the reduced with tested not before I suppose?